Sources/NIO/Selector.swift

```diff
@@ -139,22 +204,32 @@ final class Selector<R: Registration> {
 #endif
     }
 
-#if os(Linux)
 
-    private static func toEpollEvents(interested: IOEvent) -> UInt32 {
+#if os(Linux)
+    private static func toEpollEvents(interested: SelectorEventSet) -> UInt32 {
         // Also merge EPOLLRDHUP in so we can easily detect connection-reset
-        switch interested {
-        case .read:
-            return Epoll.EPOLLIN.rawValue | Epoll.EPOLLERR.rawValue | Epoll.EPOLLRDHUP.rawValue
-        case .write:
-            return Epoll.EPOLLOUT.rawValue | Epoll.EPOLLERR.rawValue | Epoll.EPOLLRDHUP.rawValue
-        case .all:
-            return Epoll.EPOLLIN.rawValue | Epoll.EPOLLOUT.rawValue | Epoll.EPOLLERR.rawValue | Epoll.EPOLLRDHUP.rawValue
-        case .none:
-            return Epoll.EPOLLERR.rawValue
+
+        var filter: UInt32 = 0
+        let epollFilters = EPollFilterSet(selectorEventSet: interested)
+        if epollFilters.contains(.error) {
+            filter |= Epoll.EPOLLERR.rawValue
+        }
+        if epollFilters.contains(.hangup) {
+            filter |= Epoll.EPOLLHUP.rawValue
+        }
+        if epollFilters.contains(.input) {
+            filter |= Epoll.EPOLLIN.rawValue
+        }
+        if epollFilters.contains(.output) {
+            filter |= Epoll.EPOLLOUT.rawValue
         }
+        if epollFilters.contains(.readHangup) {
+            filter |= Epoll.EPOLLRDHUP.rawValue
+        }
```

Comment:
Should this logic move to an extension on `EPollFilterSet`?

Reply:
@Lukasa ideally yes but unfortunately we don't have those constants available on all platforms :|